### PR TITLE
LSF support

### PR DESCRIPTION
**Pull request by maxcapodi78**

### CODE REVIEW

File: pyaedt/desktop.py [hunk 1]
```diff
@@ -65,6 +65,42 @@
     settings.use_grpc_api = True
 
 
+def launch_aedt_in_lsf(non_graphical, port):  # pragma: no cover
+    """Launch AEDT in Lsf in GRPC mode."""
```
Comment (PipKat) [inline pyaedt/desktop.py:69]:
```suggestion
    """Launch AEDT in LSF in GRPC mode."""
```
Changing to all capital letters as most of the entries seem to put it there. Should we define what LSF stands for?

File: pyaedt/generic/general_methods.py [hunk 2]
```diff
@@ -1575,6 +1575,57 @@ def __init__(self):
...
         self._edb_dll_path = None
+        self._lsf_num_cores = 2
+        self._lsf_ram = 1000
+        self._use_lsf_scheduler = False
+        self._lsf_aedt_command = "ansysedt"
+        self._lsf_timeout = 3600
+
+    @property
+    def use_lsf_scheduler(self):
+        """Either to use or not LSF Scheduler. Valid only on linux systems running LSF Scheduler."""
```
Comment (PipKat) [inline pyaedt/generic/general_methods.py:1586]:
```suggestion
        """Whether to use LSF Scheduler. This attribute is valid only on Linux systems running LSF Scheduler."""
```

File: pyaedt/generic/general_methods.py [hunk 3]
```diff
@@ -1575,6 +1575,57 @@ def __init__(self):
...
+        """Get or set the number of LSF cores. Valid only on linux systems running LSF Scheduler."""
+        return self._lsf_num_cores
+
+    @lsf_num_cores.setter
+    def lsf_num_cores(self, val):
+        self._lsf_num_cores = int(val)
+
+    @property
+    def lsf_ram(self):
+        """Get or set the RAM allocated of LSF Job. Valid only on linux systems running LSF Scheduler."""
```
Comment (PipKat) [inline pyaedt/generic/general_methods.py:1614]:
```suggestion
        """Get or set the RAM allocated for the LSF job. This attribute is valid
        only on Linux systems running LSF Scheduler."""
```

File: pyaedt/generic/general_methods.py [hunk 4]
```diff
@@ -1575,6 +1575,57 @@ def __init__(self):
...
+        return self._lsf_ram
+
+    @lsf_ram.setter
+    def lsf_ram(self, val):
+        self._lsf_ram = int(val)
+
+    @property
+    def lsf_timeout(self):
+        """Get or set the Timeout for Interactive Session startup. Default is ``3600`` seconds."""
```
Comment (PipKat) [inline pyaedt/generic/general_methods.py:1623]:
```suggestion
        """Get or set the timeout for starting the interactive session. The default is ``3600`` seconds."""
```
Do we need this sentence added: This attribute is valid only on Linux systems running LSF Scheduler.

File: pyaedt/generic/general_methods.py [hunk 5]
```diff
@@ -1575,6 +1575,57 @@ def __init__(self):
...
+        self._lsf_aedt_command = val
+
+    @property
+    def lsf_num_cores(self):
+        """Get or set the number of LSF cores. Valid only on linux systems running LSF Scheduler."""
+        return self._lsf_num_cores
+
+    @lsf_num_cores.setter
+    def lsf_num_cores(self, val):
+        self._lsf_num_cores = int(val)
```
Comment (MaxJPRey) [inline pyaedt/generic/general_methods.py]:
```suggestion
    def lsf_num_cores(self, value):
        self._lsf_num_cores = int(value)
```

File: pyaedt/generic/general_methods.py [hunk 6]
```diff
@@ -1575,6 +1575,57 @@ def __init__(self):
...
+        self._lsf_num_cores = int(value)
+
+    @property
+    def lsf_ram(self):
+        """Get or set the RAM allocated of LSF Job. Valid only on linux systems running LSF Scheduler."""
+        return self._lsf_ram
+
+    @lsf_ram.setter
+    def lsf_ram(self, val):
+        self._lsf_ram = int(val)
```
Comment (MaxJPRey) [inline pyaedt/generic/general_methods.py]:
```suggestion
    def lsf_ram(self, value):
        self._lsf_ram = int(value)
```

File: pyaedt/generic/general_methods.py [hunk 7]
```diff
@@ -1575,6 +1575,57 @@ def __init__(self):
...
+        self._lsf_ram = int(value)
+
+    @property
+    def lsf_timeout(self):
+        """Get or set the Timeout for Interactive Session startup. Default is ``3600`` seconds."""
+        return self._lsf_timeout
+
+    @lsf_timeout.setter
+    def lsf_timeout(self, val):
+        self._lsf_timeout = int(val)
```
Comment (MaxJPRey) [inline pyaedt/generic/general_methods.py]:
```suggestion
    def lsf_timeout(self, value):
        self._lsf_timeout = int(value)
```

File: pyaedt/generic/general_methods.py [hunk 8]
```diff
@@ -1575,6 +1575,57 @@ def __init__(self):
...
+
+    @property
+    def lsf_aedt_command(self):
+        """Get or set the ansysedt command to launch. Default is ``"ansysedt"``.
+        Valid only on linux systems running LSF Scheduler."""
+        return self._lsf_aedt_command
+
+    @lsf_aedt_command.setter
+    def lsf_aedt_command(self, val):
+        self._lsf_aedt_command = val
```
Comment (MaxJPRey) [inline pyaedt/generic/general_methods.py:1601]:
```suggestion
    def lsf_aedt_command(self, value):
        self._lsf_aedt_command = value
```

File: pyaedt/generic/general_methods.py [hunk 9]
```diff
@@ -1575,6 +1575,57 @@ def __init__(self):
...
+        self._lsf_timeout = 3600
+
+    @property
+    def use_lsf_scheduler(self):
+        """Either to use or not LSF Scheduler. Valid only on linux systems running LSF Scheduler."""
+        return self._use_lsf_scheduler
+
+    @use_lsf_scheduler.setter
+    def use_lsf_scheduler(self, val):
+        self._use_lsf_scheduler = val
```
Comment (MaxJPRey) [inline pyaedt/generic/general_methods.py:1591]:
```suggestion
    def use_lsf_scheduler(self, value):
        self._use_lsf_scheduler = value
```

**Review by MaxJPRey — APPROVED** [review]
LGTM.

**Pull request merged**